### PR TITLE
Make SvgeElement Parent property writable

### PR DESCRIPTION
**Pull request by wieslawsoltes**
#### Reference Issue

This is followup from discussion in https://github.com/vvvv/SVG/issues/527

#### What does this implement/fix? Explain your changes.

Makes `SvgeElement` `Parent` property writable. This is needed for implementing different rendering systems as it is necessary to set `Parent` to apply inherited styles (e.g. for `SvgUse` element). 

#### Any other comments?

Sample code I use in my rendering system implementation:
```C#
case SvgUse svgUse:
    {
        var svgVisualElement = GetReference<SvgVisualElement>(svgUse, svgUse.ReferencedElement);
        if (svgVisualElement != null)
        {
            var parent = svgUse.Parent;
            svgVisualElement.Parent = svgUse;
            svgVisualElement.InvalidateChildPaths();

            canvas.Save();

            float x = svgUse.X.ToDeviceValue(null, UnitRenderingType.Horizontal, svgUse);
            float y = svgUse.Y.ToDeviceValue(null, UnitRenderingType.Vertical, svgUse);
            canvas.Translate(x, y);

            if (svgVisualElement is SvgSymbol)
            {
                DrawSymbol(canvas, svgVisualElement as SvgSymbol);
            }
            else
            {
                Draw(canvas, svgVisualElement);
            }

            svgVisualElement.Parent = parent;

            canvas.Restore();
        }
    }
    break;
```

### CODE REVIEW

**Comment by wieslawsoltes** [comment]
For reference: https://github.com/vvvv/SVG/issues/527#issuecomment-515205728

**Comment by wieslawsoltes** [comment]
For reference, this is code I have currently, that uses reflection:
```C#
case SvgUse svgUse:
    {
        var svgVisualElement = GetReference<SvgVisualElement>(svgUse, svgUse.ReferencedElement);
        if (svgVisualElement != null)
        {
            var parent = svgUse.Parent;
            //svgVisualElement.Parent = svgUse;
            var _parent = svgUse.GetType().GetField("_parent", BindingFlags.NonPublic | BindingFlags.Instance);
            _parent.SetValue(svgVisualElement, svgUse);
            svgVisualElement.InvalidateChildPaths();

            canvas.Save();
            DrawOpacity(canvas, svgElement, size);
            canvas.Save();
            Transform(canvas, svgElement.Transforms);

            float x = svgUse.X.ToDeviceValue(null, UnitRenderingType.Horizontal, svgUse);
            float y = svgUse.Y.ToDeviceValue(null, UnitRenderingType.Vertical, svgUse);
            canvas.Translate(x, y);

            var ew = svgUse.Width.ToDeviceValue(null, UnitRenderingType.Horizontal, svgUse);
            var eh = svgUse.Height.ToDeviceValue(null, UnitRenderingType.Vertical, svgUse);
            if (ew > 0 && eh > 0)
            {
                var _attributes = svgVisualElement.GetType().GetField("_attributes", BindingFlags.NonPublic | BindingFlags.Instance);
                var attributes = _attributes.GetValue(_attributes) as SvgAttributeCollection;
                var viewBox = attributes.GetAttribute<SvgViewBox>("viewBox");
                //var viewBox = svgVisualElement.Attributes.GetAttribute<SvgViewBox>("viewBox");
                if (viewBox != SvgViewBox.Empty && Math.Abs(ew - viewBox.Width) > float.Epsilon && Math.Abs(eh - viewBox.Height) > float.Epsilon)
                {
                    var sw = ew / viewBox.Width;
                    var sh = eh / viewBox.Height;
                    canvas.Translate(sw, sh);
                }
            }

            if (svgVisualElement is SvgSymbol)
            {
                DrawSymbol(canvas, svgVisualElement as SvgSymbol, size);
            }
            else
            {
                Draw(canvas, svgVisualElement, size);
            }

            //svgVisualElement.Parent = parent;
            _parent.SetValue(svgVisualElement, parent);

            canvas.Restore();
            canvas.Restore();
        }
    }
    break;
```

**Comment by wieslawsoltes** [comment]
Another place where I can not access internal state necessary for rendering:
```C#
var _attributes = svgVisualElement.GetType().GetField("_attributes", BindingFlags.NonPublic | BindingFlags.Instance);
var attributes = _attributes.GetValue(_attributes) as SvgAttributeCollection;
var viewBox = attributes.GetAttribute<SvgViewBox>("viewBox");
//var viewBox = svgVisualElement.Attributes.GetAttribute<SvgViewBox>("viewBox");
```
I would also like to have access to `SvgElement.Attributes.GetAttribute<T>`

File: Source/SvgElement.cs [hunk 1]
```diff
@@ -281,7 +282,7 @@ public virtual SvgDocument OwnerDocument
         /// <summary>
         /// Gets a collection of element attributes.
         /// </summary>
-        protected internal virtual SvgAttributeCollection Attributes
+        public virtual SvgAttributeCollection Attributes
```
Comment (H1Gdev) [inline Source/SvgElement.cs]: Why do you need this change ?
Attribute properties are public.

Comment (wieslawsoltes) [reply Source/SvgElement.cs]: Not all attributes are available in this way. Especially when using `use` element.

My use case, based on `SvgUse` code:
https://github.com/wieslawsoltes/Draw2D/blob/3939471702dc7a11a4e9e89455c5c3d2cfcc71ea/svg/Svg.Skia/Elements/Use.cs#L37

Comment (H1Gdev) [reply Source/SvgElement.cs]: I don't think so.
Should make `SvgUse` class better.

**Pull request closed without merging**